src/main/java/org/recommendation/service/UnWatchedMovieFilter.java

```diff
@@ -0,0 +1,11 @@
+package org.recommendation.service;
+
+import org.recommendation.model.User;
+
+
+public class UnWatchedMovieFilter implements Filter {
+    @Override
+    public boolean satisfy(final String movieId, final User user) {
+        return !user.checkIfMovieRatedByUser(movieId);
```

Comment:
null pointer exception in case of user is null

Reply:
for these type of handling we can add @NotNull with the argument

Reply:
done